### PR TITLE
Fix lints flagged by llvm 17

### PR DESCRIPTION
**Pull request by bifurcation**
Work on #367 indicated that the latest `llvm` added some lints that were hitting on our code.  This PR updates to address those lints:

* Forwarding parameters are either converted to `const&` or passed to `std::forward`
* GREASE is refactored to operate on non-const references instead of moving in and out
* Some lints are ignored (e.g., misc-include-cleaner)

### CODE REVIEW

**Comment by glhewett** [comment]
fixes #380

File: .clang-tidy [hunk 1]
```diff
@@ -20,12 +20,15 @@ Checks:            '*,
...
+                    -llvmlibc-inline-function-decl,
                     -llvmlibc-restrict-system-libc-headers,
+                    -misc-include-cleaner,
+                    -misc-no-recursion,
                     -misc-non-private-member-variables-in-classes,
                     -misc-use-anonymous-namespace,
-                    -misc-no-recursion,
                     -modernize-use-nodiscard,
                     -modernize-use-trailing-return-type,
+                    -performance-avoid-endl,
```
Comment (glhewett) [inline .clang-tidy:31]: Some of these clang-tidy exclusions look handy, but I am sure they are a pain.  I am not going to look at every one.

**Review by glhewett — APPROVED** [review]
LGTM

File: .clang-tidy (hunk 1 above)
Comment (bifurcation) [reply .clang-tidy:31]: We can come back and review them later.

**Comment by bifurcation** [comment]
Merging despite macOS hang, known issue.

**Pull request merged**